giskard/rag/knowledge_base.py

```diff
@@ -185,13 +507,35 @@ def from_pandas(cls, df: pd.DataFrame, columns: Optional[Sequence[str]] = None,
         """
         return cls(data=df, columns=columns, **kwargs)
 
+    def _store_embeddings_inst(self, embeddings_inst: np.ndarray):
+        self._embeddings_inst = embeddings_inst
+        for doc, emb in zip(self._documents, self._embeddings_inst):
+            doc.embeddings = emb
+
+    def _store_reduced_embeddings_inst(self, reduced_embeddings_inst: np.ndarray):
+        self._reduced_embeddings_inst = reduced_embeddings_inst
+        for doc, emb in zip(self._documents, self._reduced_embeddings_inst):
+            doc.reduced_embeddings = emb
```

Comment:
I think these are nice methods that we could re-use also within the other serialisation methods like `from_pandas` to ease the flow of re-using embeddings.

Reply:
Similarly, we could consider making them public methods.